File: Interpretor2/src/Repository/Repository.java
package Repository;
import java.io.BufferedWriter;
import java.io.FileWriter;
import java.io.PrintWriter;
import java.util.ArrayList;
import java.util.List;

import Model.*;
public class Repository implements IRepository {
	List<PrgState> list;
	int index;
	String  logFilePath;
	public Repository(PrgState s,String path)
	{
		this.list=new ArrayList<PrgState>();
		this.list.add(s);
		this.index=0;
		this.logFilePath=path;
	}
	public PrgState getState(){
		return this.list.get(this.index);
	}
	public void logPrgStateExec() throws Exception
	{
		try{	
		PrintWriter logFile = new PrintWriter(new BufferedWriter(new FileWriter(this.logFilePath, true)));
			//System.out.println(this.logFilePath);
			logFile.append(this.list.get(this.index).toString());
			logFile.flush();
			//logFile.close();
		}
		catch(Exception e)
		{
			
		}
	}
}

File: MAP/Interpreter C#/Interpreter/Repository.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Interpreter.Model;

namespace Interpreter
{
    namespace Repo
    {
        
        class Repository:IRepository
        {
            List<PrgState> lst;
            int index;
            string pathFile;

            public Repository(PrgState p,string path)
            {
                this.lst = new List<PrgState>();
                this.lst.Add(p);
                this.index = 0;
                this.pathFile = path;
            }

            public List<PrgState> Programs
            {
                set
                {
                    this.lst = value;
                    this.index = 0;
                }
                get { return this.lst; }

            }

            PrgState IRepository.getCrtPrg()
            {
                return this.lst.ElementAt(this.index);
            }

            void IRepository.logPrgStateExec()
            {
                using (System.IO.StreamWriter file =
            new System.IO.StreamWriter(this.pathFile,true))
                {
                    foreach (PrgState program in this.lst)
                    {
                        file.WriteLine(program.ToString());
                        file.WriteLine("\n----------------------------------------------------------\n");
                       
                    }
                }

            }
        }
    }
}
